### PR TITLE
Doc/install/aws/uninstall: Describe CLI destroy of GUI-made cluster

### PR DESCRIPTION
**Pull request by joshix**
remove CF references
Add steps for terraform destroy using the provided tools and config
from installer or assets.zip.

(does not include CLI User-level instructions update in each of the *-terraform.md docs.)

### CODE REVIEW

File: Documentation/install/aws/uninstall.md [hunk 1]
```diff
@@ -1,43 +1,50 @@
 # Uninstall
```
Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: I think Uninstall Tectonic, please.

File: Documentation/install/aws/uninstall.md [hunk 2]
```diff
@@ -1,43 +1,50 @@
...
 
-## Delete LoadBalancer services and CloudFormation StackFormation
+## Deleting a cluster created by Tectonic Installer
 
-### Delete Kubernetes LoadBalancer services
+Clusters created with the graphical Tectonic Installer can be deleted with a few commands using tools included in either the Installer download, or the cluster's `assets.zip` file. Recent users of Tectonic Installer will have a `tectonic` directory containing the necessary tools and configuration, often found in the system `Downloads` folder.
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: I feel like this is unclear.. It does not clearly say what's the assets.zip and where do I get it. the `Installer download` is a big unclear too.

Then like I dont clearly understand how I will have a magical `tectonic` folder in `Downloads`.

File: Documentation/install/aws/uninstall.md [hunk 3]
```diff
@@ -1,43 +1,50 @@
...
 
-An attempt to delete a CloudFormation stack on which a Tectonic cluster has been deployed may hang if there are existing Kubernetes `LoadBalancer` services in the cluster.
+To destroy a given Tectonic Cluster and the resources provisioned for it by Tectonic Installer, open a Terminal and change directories to the cluster state directory for the target cluster within the Tectonic Installer directory hierarchy. The cluster state directory is named with the name given to the cluster in the install process, suffixed with the date and time of the install.
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: `cluster state directory` --> not sure they will know what that is at first. Would be better to say that they gotta change directory to what they unzipped from assets.zip, that they downloaded.

Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: ...Tectonic cluster ...

open a Terminal and navigate to the cluster state directory for the target cluster within the Tectonic Installer directory. This directory is named with the name given the cluster during the install process, suffixed with ...

File: Documentation/install/aws/uninstall.md [hunk 4]
```diff
@@ -1,43 +1,50 @@
...
 
-Kubernetes [`LoadBalancer` services][k8s-lb] expose cluster facilities by manipulating the AWS Elastic Load Balancer (ELB) API. These ELBs are created dynamically, after the creation of the CloudFormation stack that defines the cluster nodes and network environment. Such ELBs that are not part of the CloudFormation stack, but that are attached to a CloudFormation-managed subnet, can cause attempts to delete the CloudFormation stack to fail.
+```bash
+# Replace <os> with one of darwin, linux, or windows
+# Replace <CLUSTERNAME> with a string like mytectonic_2017-05-03_11-41-02
+$ cd tectonic/tectonic-installer/<os>/clusters/<CLUSTERNAME>
+$ export PATH=$PATH:$(pwd)/../..	# Add Installer's terraform binary to PATH
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: I'd reverse the two terms. Otherwise it might use their outdated/newer TerraForm.

File: Documentation/install/aws/uninstall.md [hunk 5]
```diff
@@ -1,43 +1,50 @@
...
+$ TERRAFORM_CONFIG=.terraformrc terraform destroy --force
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: To be safe, I'd use `TERRAFORM_CONFIG=$(pwd)/.terraformrc`

File: Documentation/install/aws/uninstall.md [hunk 6]
```diff
@@ -1,43 +1,50 @@
...
+```
 
-Delete all services of `type=LoadBalancer` with `kubectl delete` or in Tectonic Console to avoid this issue.
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: This still need to be here otherwise the route53 zone will fail to be deleted.

File: Documentation/install/aws/uninstall.md [hunk 7]
```diff
@@ -1,43 +1,50 @@
...
+Terraform will ask for the target AWS region where the cluster was created. AWS regions have names like `us-west-1`.
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: No, https://github.com/coreos/tectonic-installer/pull/509 will fix that.

Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: Note that they still need to export their AWS credentials for now!

File: Documentation/install/aws/uninstall.md [hunk 8]
```diff
@@ -1,43 +1,50 @@
...
 
-For example, to delete *every* LoadBalancer service in a cluster with `kubectl`:
+If using a a state directory extracted from `assets.zip`, add a recent Tectonic Installer's `terraform` binary to the shell's $PATH before changing to the directory extracted from `assets.zip` and invoking `terraform destroy` as above.
 
-```sh
-$ kubectl delete services -l type=LoadBalancer
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: still needed

File: Documentation/install/aws/uninstall.md [hunk 9]
```diff
@@ -1,43 +1,50 @@
...
+$ cd tectonic/tectonic-installer/<os>/clusters/<CLUSTERNAME>
+$ export PATH=$PATH:$(pwd)/../..	# Add Installer's terraform binary to PATH
+$ TERRAFORM_CONFIG=.terraformrc terraform destroy --force
+```
 
-Delete all services of `type=LoadBalancer` with `kubectl delete` or in Tectonic Console to avoid this issue.
+Terraform will ask for the target AWS region where the cluster was created. AWS regions have names like `us-west-1`.
 
-For example, to delete *every* LoadBalancer service in a cluster with `kubectl`:
+If using a a state directory extracted from `assets.zip`, add a recent Tectonic Installer's `terraform` binary to the shell's $PATH before changing to the directory extracted from `assets.zip` and invoking `terraform destroy` as above.
```
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: a a

Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: not needed if they export the PATH as done above, right?

File: Documentation/install/aws/uninstall.md (hunk 8 above)
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: got it, it's down below.

File: Documentation/install/aws/uninstall.md (hunk 6 above)
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: got it, it's down below.

File: Documentation/install/aws/uninstall.md [hunk 10]
```diff
@@ -1,43 +1,50 @@
...
 
-If any errors occur be sure to delete any additional resources you may have added that reference resources in the StackFormation.
+Such resources include ELBs manually created with endpoints in the cluster subnet, existing or manually added Route53 DNS records, Elastic Block Store (EBS) volumes that were attached to nodes or pods after installing Tectonic, and any other resources associated with Tectonic's Virtual Private Cloud (VPC) but not created by the Tectonic installer. Delete these resources by using their AWS service's UI or API.
 
 ## Reinstall
 
-Use the latest [Tectonic AWS][install-aws] documentation to create a new cluster and deploy your applications and services. If your new cluster will be similar to a previous cluster, you may supply the `tectonic.progress` "progress file" found in your existing [assets bundle][assets] to base the installation on that configuration.
+Use the latest [Tectonic AWS][install-aws] documentation to create a new cluster and deploy your applications and services.
```
Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: I would either remove this section, or change it to more clearly state - use the assets to build a similar cluster:

## Reinstall using previous configuration

Use the `tectonic.progress` progress file found in your existing [assets bundle][assets] to create a new cluster based on configuration options from an previous cluster. Follow the [AWS: Installation][install-aws] guide to create a new cluster and deploy your applications and services.

File: Documentation/install/aws/uninstall.md (hunk 3 above)
Comment (joshix) [inline Documentation/install/aws/uninstall.md]: yeah I like your clearer thinking here. I'm going to rewrite it quickly to favor the assets.zip rather than a leftover installer

File: Documentation/install/aws/uninstall.md (hunk 10 above)
Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: we don't provide the progress file in there.

Comment (Quentin-M) [inline Documentation/install/aws/uninstall.md]: it can be only be saved from the GUI, but this is more like a dev feature than anything else

File: Documentation/install/aws/uninstall.md (hunk 3 above)
Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: now that i've done it, I understand. will take a look at your new version, when it's ready.

**Comment by Quentin-M** [comment]
they won't have to delete route53 records anymore, but probably still the ELBs and ECS resources!

**Comment by joshix** [comment]
@Quentin-M Delete reference to route53 records

File: Documentation/install/aws/uninstall.md [hunk 11]
```diff
@@ -1,43 +1,65 @@
...
 
-Kubernetes [`LoadBalancer` services][k8s-lb] expose cluster facilities by manipulating the AWS Elastic Load Balancer (ELB) API. These ELBs are created dynamically, after the creation of the CloudFormation stack that defines the cluster nodes and network environment. Such ELBs that are not part of the CloudFormation stack, but that are attached to a CloudFormation-managed subnet, can cause attempts to delete the CloudFormation stack to fail.
+Recent users of Tectonic Installer will have downloaded a release `.tar.gz` file, extracted a `tectonic` directory from it, and executed the installation from there. Locate the directory where Tectonic Installer was extracted and executed.
 
-Delete all services of `type=LoadBalancer` with `kubectl delete` or in Tectonic Console to avoid this issue.
+#### Export AWS credentials
 
-For example, to delete *every* LoadBalancer service in a cluster with `kubectl`:
+First, open a terminal and set AWS credentials for the destroy operation. The AWS credentials used to perform the installation are recommended. Replace `<ACCESSKEYID>` and `<SECRETACCESSKEY>` in the commands below with appropriate values for the AWS account.
```
Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: These keys were generated as part of the AWS setup process, as described in [Creating AWS access keys][https://coreos.com/tectonic/docs/latest/tutorials/creating-aws.html].

File: Documentation/install/aws/uninstall.md [hunk 12]
```diff
@@ -1,43 +1,65 @@
...
 
-```sh
-$ kubectl delete services -l type=LoadBalancer
+```bash
+$ export AWS_ACCESS_KEY_ID=<ACCESSKEY>
```
Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: <ACCESSKEYID>

File: Documentation/install/aws/uninstall.md (hunk 11 above)
Comment (joshix) [inline Documentation/install/aws/uninstall.md]: pass on expanding this here

Comment (zbwright) [inline Documentation/install/aws/uninstall.md]: I think you need to run this command from within the <os> folder for the installer? It looks like you can't just run it from anywhere.
unless I'm doing it wrong.

File: Documentation/install/aws/uninstall.md (hunk 12 above)
Comment (joshix) [inline Documentation/install/aws/uninstall.md]: fixed

File: Documentation/install/aws/uninstall.md [hunk 13]
```diff
@@ -1,43 +1,65 @@
...
+Next, navigate to the cluster state directory written to the extracted `tectonic` directory during the install process. Cluster state directories are stored beneath the same parent directory as the `installer` and `terraform` binaries, in a child directory called `clusters`. Each state directory beneath `clusters` is named by a cluster's name suffixed with the date and time of the install.
 
-Such resources include ELBs manually created with endpoints in the CloudFormation subnet, Elastic Block Store (EBS) volumes that were attached to nodes or pods after installing Tectonic, and any other resources associated with Tectonic's Virtual Private Cloud (VPC) but not created by the Tectonic installer. Delete these resources by using their AWS service's UI or API.
+```bash
+# Replace <os> with one of darwin, linux, or windows
+# Replace <CLUSTERNAME> with a string like mytectonic_2017-05-03_11-41-02
+$ cd tectonic/tectonic-installer/<os>/clusters/<CLUSTERNAME>
+$ export PATH=$(pwd)/../..:$PATH	# Add Installer's terraform binary to PATH
+$ TERRAFORM_CONFIG=$(pwd)/.terraformrc terraform destroy --force
+```
```
Comment (robszumski) [inline Documentation/install/aws/uninstall.md]: Terraform only destroys objects that it has created. Don't worry blah blah...

Comment (joshix) [inline Documentation/install/aws/uninstall.md]: https://github.com/coreos/tectonic-installer/pull/518/files#diff-218c509e40308277813cb75996ba94d6R38

File: Documentation/install/aws/uninstall.md [hunk 14]
```diff
@@ -1,43 +1,65 @@
...
+Next, navigate to the cluster state directory written to the extracted `tectonic` directory during the install process. Cluster state directories are stored beneath the same parent directory as the `installer` and `terraform` binaries, in a child directory called `clusters`. Each state directory beneath `clusters` is named by a cluster's name suffixed with the date and time of the install.
 
-Such resources include ELBs manually created with endpoints in the CloudFormation subnet, Elastic Block Store (EBS) volumes that were attached to nodes or pods after installing Tectonic, and any other resources associated with Tectonic's Virtual Private Cloud (VPC) but not created by the Tectonic installer. Delete these resources by using their AWS service's UI or API.
+```bash
+# Replace <os> with one of darwin, linux, or windows
+# Replace <CLUSTERNAME> with a string like mytectonic_2017-05-03_11-41-02
+$ cd tectonic/tectonic-installer/<os>/clusters/<CLUSTERNAME>
+$ export PATH=$(pwd)/../..:$PATH	# Add Installer's terraform binary to PATH
+$ TERRAFORM_CONFIG=$(pwd)/.terraformrc terraform destroy --force
```
Comment (robszumski) [inline Documentation/install/aws/uninstall.md]: any reason to combine this env var on one line, but not the path one? might be more understandable

File: Documentation/install/aws/uninstall.md [hunk 15]
```diff
@@ -1,43 +1,65 @@
...
+```
```
Comment (robszumski) [inline Documentation/install/aws/uninstall.md]: might be nice here to include a little bit of sample output:

```
$ TERRAFORM_CONFIG=$(pwd)/.terraformrc terraform destroy --force
provider.aws.region
  The region where AWS operations will take place. Examples
  are us-east-1, us-west-2, etc.

  Default: us-east-1
  Enter a value: us-west-1

tls_private_key.ingress: Refreshing state... (ID: 38aaae42623d255797e70602cf81b27574496fdf)
tls_private_key.identity-server: Refreshing state... (ID: 4b4d568965570a3d88acb7ab8121cb6afa587fae)
tls_private_key.identity-client: Refreshing state... (ID: 5dc4dd159f5de9318a3cd9ca4c13c89801f749b8)
template_dir.bootkube-bootstrap: Refreshing state... (ID: 7b3b54f349160ff1eefad8ce1a01c4d62b02cae8)
random_id.console_secret: Refreshing state... (ID: 74U0bR3MQMztiz1sasRRMw)
tls_private_key.kube-ca: Refreshing state... (ID: a93dbf10de4297cbb03f65dfbc551d3abc0280c3)
...snip....
module.vpc.aws_security_group.master: Destroying... (ID: sg-33693754)
module.vpc.aws_security_group.worker: Destroying... (ID: sg-0b66386c)
module.vpc.aws_security_group.worker: Destruction complete
module.vpc.aws_security_group.master: Destruction complete

Destroy complete! Resources: 41 destroyed.
```

File: Documentation/install/aws/uninstall.md (hunk 13 above)
Comment (robszumski) [inline Documentation/install/aws/uninstall.md]: double "not" is kinda hard to parse

File: Documentation/install/aws/uninstall.md (hunk 14 above)
Comment (joshix) [inline Documentation/install/aws/uninstall.md]: The idea was avoiding doing any exports, and thus leaving the environment as we found it. However, that was before I added the AWS creds so is maybe less compelling now

File: Documentation/install/aws/uninstall.md (hunk 15 above)
Comment (joshix) [inline Documentation/install/aws/uninstall.md]: Seems reasonable -- I'll mock in output that doesn't ask for region

**Comment by joshix** [comment]
ok to test

**Comment by joshix** [comment]
ok to test
